src/client/testing/testController/controller.ts

```diff
@@ -143,6 +148,7 @@ export class PythonTestController implements ITestController, IExtensionSingleAc
             if (settings.testing.pytestEnabled) {
                 await this.pytest.refreshTestData(this.testController, uri, this.refreshCancellation.token);
             } else if (settings.testing.unittestEnabled) {
+                // TODO: Use new test discovery mechanism
                 await this.unittest.refreshTestData(this.testController, uri, this.refreshCancellation.token);
```

Comment:
New discovery would look like this:

```suggestion
                const testAdapter =
                    this.testAdapters.get(uri) || (this.testAdapters.values().next().value as WorkspaceTestAdapter);
                testAdapter.discoverTests(
                    this.testController,
                    this.refreshCancellation.token,
                    this.testAdapters.size > 1,
                    this.workspaceService.workspaceFile?.fsPath || Common.workspace(),
                );
```